src/librustc_trans/trans/consts.rs

```diff
@@ -566,12 +610,20 @@ fn const_expr_unadjusted<'a, 'tcx>(cx: &CrateContext<'a, 'tcx>,
           }
 
           ast::ExprIndex(ref base, ref index) => {
+              match const_eval::eval_const_expr_partial(cx.tcx(), &e, None) {
```

Comment:
an alternative would be to check if `base` points to a static or a constant and then decide on what to do. Any thoughts?